File: test/python/test_read_sleep_stages.py

from sys import platform
import sys
sys.path.insert(0, '')

if platform == "linux" or platform == "linux2":
    import build.HarmonieReader as HarmonieReader
elif platform == "darwin":
    import build.HarmonieReader as HarmonieReader
elif platform == "win32":
    import build.Release.HarmonieReader as HarmonieReader
    

reader = HarmonieReader.HarmonieReader()
print("Test: test_read_sleep_stages")

filename = ""
if filename == "":
    print("ERROR: No filename specified")
    quit()

print(f"Opening file... :{filename}")
res = reader.open_file(filename)

if (not res):
    print(f"ERROR Failed to open the file:{filename}")
    quit()
else:
    print("SUCCESS")

print(f"Reading sleep stages")
sleep_stages = reader.get_sleep_stages()
print(f"Sleep stages count:{len(sleep_stages)}")
for s in sleep_stages:
    print(f"{s.sleep_stage}\t {s.start_time}\t {s.duration}")

File: test/cpp/test_read_sleep_stages.cpp
// Your First C++ Program

#include <iostream>
#include "../src/harmonie_reader.h"

using namespace Harmonie;

int main() {
    HarmonieReader reader;
    std::cout << "Test: test_read_sleep_stages" << std::endl;

    std::string filename = "";
    if (filename == "") {
        std::cout << "ERROR: No filename specified" << std::endl;
        return 0;
    }

    std::cout << "Opening file... :" << filename << std::endl;
    bool res = reader.openFile(filename);

    if (!res) {
        std::cout << "ERROR Failed to open the file:" << filename << std::endl;
        return 0;
    } else {
        std::cout << "SUCCESS" << std::endl;
    }

    std::vector<HarmonieReader::SLEEP_STAGE> sleepStages = reader.getSleepStages();

    std::cout << "SleepStages:" << sleepStages.size() << std::endl << std::endl;
    for (int i=0; i<sleepStages.size(); i++) {
        std::cout << sleepStages[i].sleepStage << " " << sleepStages[i].startTime << " " << sleepStages[i].duration << std::endl;
    }

    std::cout << "DONE" << std::endl;
    return 0;
}
